tools/import-image-tests

```diff
@@ -63,40 +60,78 @@ def load_test(name, fullpath) -> dict:
     return test
 
 
-def export_test(test, target, manifest_only):
+def export_test(test: Dict, target: Union[str, pathlib.Path], manifest_only: bool) -> None:
     name = test["id"]
     fullpath = os.path.join(target, name + ".json")
     if manifest_only and os.path.exists(fullpath):
         # load existing
         with open(fullpath, "r", encoding="utf-8") as f:
             base = json.load(f)
-        # only replace the manifest in the base
         base["manifest"] = test["manifest"]
         test = base
     with open(fullpath, "w", encoding="utf-8") as f:
         json.dump(test, f, indent=2)
 
 
-def report_import(test, verbosity=1):
+def report_ignoring(name: str, reason: str, verbosity: int = 1) -> None:
     if verbosity < 1:
         return
 
     if verbosity == 1:
         print(".", end="", flush=True)
     else:
-        print(f"{GREEN} OK  {RESET} {test['id']}", flush=True)
+        print(f"{YELLOW} SKIP  {reason} {RESET}{name}", flush=True)
 
 
-def report_failure(name, reason, verbosity=1):
+def report_import(test: Dict, verbosity: int = 1) -> None:
     if verbosity < 1:
         return
 
     if verbosity == 1:
-        print(f"{RED}F{RESET}", end="", flush=True)
+        print(".", end="", flush=True)
     else:
-        print(f"{RED}FAIL{RESET} {name}: {reason}")
+        print(f"{GREEN} OK   {RESET} {test['id']}", flush=True)
 
 
+def report_failure(name: str, reason: str, verbosity: int = 1) -> None:
+    if verbosity < 1:
+        return
+
+    if verbosity == 1:
+        print(f"{RED}F{RESET}", end="", flush=True)
+    else:
+        print(f"{RED}FAIL {RESET} {name}: {reason}")
+
+
+def parse_distro(ci_file: Union[str, pathlib.Path]) -> Dict[str, List[str]]:
+    filters: Dict[str, List[str]] = {}
+    if ci_file:
```

Comment:
Just a side node, I think an early return makes the code more readable because you can get rid of a level of indention.